Test/AwsCommonRuntimeKitTests/XCBaseTestCase.swift

```diff
@@ -35,20 +38,44 @@ extension XCTestCase {
     }
 
     func skipIfiOS() throws {
-        if #available(iOS 10, *) {
+        #if os(iOS)
             throw XCTSkip("Skipping test on iOS")
-        }
+        #endif
     }
 
     func skipifmacOS() throws {
-        if #available(macOS 10.14, *) {
+        #if os(macOS)
             throw XCTSkip("Skipping test on macOS")
-        }
+        #endif
     }
 
     func skipIfLinux() throws {
         #if os(Linux)
             throw XCTSkip("Skipping test on linux")
         #endif
     }
+
+    func skipIfwatchOS() throws {
+        #if os(watchOS)
+            throw XCTSkip("Skipping test on watchOS")
+        #endif
+    }
+
+    func skipIftvOS() throws {
+        #if os(tvOS)
+            throw XCTSkip("Skipping test on tvOS")
+        #endif
+    }
+}
+
+extension XCTestCase {
+    /// Return the environment variable value, or Skip the test if env var is not set.
+    func GetEnvironmentVarOrSkip(environmentVarName name: String) throws -> String? {
+        let result = ProcessInfo.processInfo.environment[name]
+        guard result != nil else {
+            try skipTest(message: "Skipping test because environment is not configured properly.")
+            return nil
+        }
+        return result
+    }
```

Comment:
* Why create a separate extension block? We can make this part of the same extension block as the `skipIf*` functions. 
* function name should start with lowercase.
* We should avoid nullable types as much as possible.
* I think it would be better to just get rid of `skipTest` function and use `throw XCTSkip(message)` directly.

```suggestion
    func getEnvironmentVarOrSkipTest(environmentVarName name: String) throws -> String {
        guard let result = ProcessInfo.processInfo.environment[name] else {
            throw XCTSkip(message: "Skipping test because environment is not configured properly.")
        }
        return result
    }
```